autoPyTorch/api/tabular_regression.py

```diff
@@ -402,20 +444,57 @@ def search(
             portfolio_selection=portfolio_selection,
         )
 
+    def _get_dataset_compression_mapping(
```

Comment:
Is it a good idea to move it to BaseTask?
I mean BaseTask will be messier, but it can be one idea if it duplicates in the future.
Another idea will be to create a function that returns both compressed_dataset and reduced_dtype so that we do not require the access to self.